scripts/troubleshoot/cluster-migration/migrate-legacy-to-msi-aks.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+#
+# Script to migrate AKS clusters from service principal to managed identity authentication
+#
+# Usage:
+#   1. Create a text file listing your clusters (e.g., clusters.txt):
+#      ResourceGroup,ClusterName
+#      myRG-prod,cluster-prod
+#
+#   2. Run the script:
+#      ./migrate-legacy-to-msi-aks.sh clusters.txt
+#
+
+if [ "$1" == "--help" ] || [ "$1" == "-h" ] || [ -z "$1" ]; then
+    echo "Migrates AKS clusters to managed identity authentication"
+    echo ""
+    echo "Usage: $0 <clusters-file>"
+    echo "The clusters file should contain one cluster per line in format: resource-group,cluster-name"
+    echo ""
+    echo "Example clusters file content:"
+    echo "myResourceGroup1,myCluster1"
+    echo "myResourceGroup2,myCluster2"
+    exit 0
+fi
+
+clusters_file=$1
+if [ ! -f "$clusters_file" ]; then
+    echo "[Error] Clusters file not found: $clusters_file"
+    exit 1
+fi
+
+# Clean and normalize line endings
+temp_file=$(mktemp)
+sed 's/\r$//' "$clusters_file" | sed '/^[[:space:]]*$/d' > "$temp_file" # Remove Windows-style line endings and empty lines
+mv "$temp_file" "$clusters_file"
+
+# Check login status
+if ! az account show &> /dev/null; then
+    echo "[Error] Please run 'az login' first"
+    exit 1
+fi
+
+# Arrays to track cluster status
+successful_clusters=()
+failed_clusters=()
+
+# Process each cluster
+while IFS=, read -r resource_group cluster_name || [ -n "$resource_group" ]; do
+    # Skip empty lines and trim whitespace
+    [ -z "$resource_group" ] && continue
+    resource_group=$(echo "$resource_group" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')
+    cluster_name=$(echo "$cluster_name" | sed -e 's/^[[:space:]]*//' -e 's/[[:space:]]*$//')
+
+    echo "[$(date)] Processing cluster: $cluster_name"
+
+    # Get workspace ID using exact command
+    echo "[$(date)] Getting workspace ID for $cluster_name"
+    workspace_line=$(az aks show -g "$resource_group" -n "$cluster_name" | grep -i "logAnalyticsWorkspaceResourceID")
```

Comment:
better naming is "workspace_resource_id"

Reply:
changed